File: src/model/Skill.java
package model;
/**
 * Skills named and with skill point cost.
 * 
 */
public enum Skill {
	MEDICAL (50, "Medical"),
	RIVERWORK (50, "Riverwork"),
	SHARPSHOOTING (50, "Sharpshooting"),
	BLACKSMITHING (40, "Blacksmithing"),
	CARPENTRY (40, "Carpentry"),
	FARMING (40, "Farming"),
	TRACKING (30, "Tracking"),
	BOTANY (20, "Botany"),
	COMMERCE (20, "Commerce"),
	COOKING (20, "Cooking"),
	MUSICAL (10, "Musical"),
	SEWING (10, "Sewing"),
	SPANISH (10, "Spanish"),
	NONE (0, "");
	
	private final int cost;

	private final String name;
	
	/**
	 * Sets the cost of the skill as designated
	 * @param cost The cost of the skill (defined above)
	 * @param name The name of the skill
	 */
	private Skill(int cost, String name) {
		this.cost = cost;
		this.name = name;
	}
	
	/**
	 * Returns the cost of a skill
	 * @return The cost
	 */
	public int getCost() {
		return cost;
	}
	
	/**
	 * Return the name of a skill
	 * @return The skill name
	 */
	public String getName(){
		return name;
	}
	
	public String toString() {
		return name;
	}
}

File: web/src/model/Skill.ts
export enum Skill {
  MEDICAL = "MEDICAL",
  RIVERWORK = "RIVERWORK",
  SHARPSHOOTING = "SHARPSHOOTING",
  BLACKSMITHING = "BLACKSMITHING",
  CARPENTRY = "CARPENTRY",
  FARMING = "FARMING",
  TRACKING = "TRACKING",
  BOTANY = "BOTANY",
  COMMERCE = "COMMERCE",
  COOKING = "COOKING",
  MUSICAL = "MUSICAL",
  SEWING = "SEWING",
  SPANISH = "SPANISH",
  NONE = "NONE",
}

const SKILL_DATA: Record<Skill, { cost: number; name: string }> = {
  [Skill.MEDICAL]: { cost: 50, name: "Medical" },
  [Skill.RIVERWORK]: { cost: 50, name: "Riverwork" },
  [Skill.SHARPSHOOTING]: { cost: 50, name: "Sharpshooting" },
  [Skill.BLACKSMITHING]: { cost: 40, name: "Blacksmithing" },
  [Skill.CARPENTRY]: { cost: 40, name: "Carpentry" },
  [Skill.FARMING]: { cost: 40, name: "Farming" },
  [Skill.TRACKING]: { cost: 30, name: "Tracking" },
  [Skill.BOTANY]: { cost: 20, name: "Botany" },
  [Skill.COMMERCE]: { cost: 20, name: "Commerce" },
  [Skill.COOKING]: { cost: 20, name: "Cooking" },
  [Skill.MUSICAL]: { cost: 10, name: "Musical" },
  [Skill.SEWING]: { cost: 10, name: "Sewing" },
  [Skill.SPANISH]: { cost: 10, name: "Spanish" },
  [Skill.NONE]: { cost: 0, name: "" },
};

export function getSkillCost(skill: Skill): number {
  return SKILL_DATA[skill].cost;
}

export function getSkillName(skill: Skill): string {
  return SKILL_DATA[skill].name;
}
